File: PHP/CountTheNumberOfSpecialCharactersIITest.php
<?php
/**
 * @version PHP 8.2.22
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit CountTheNumberOfSpecialCharactersIITest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("CountTheNumberOfSpecialCharactersII.php");

class CountTheNumberOfSpecialCharactersIITest extends TestCase {

    public function testCountTheNumberOfSpecialCharactersII() {
        $obj = new CountTheNumberOfSpecialCharactersII();
        //Test 1
        $want = 3;
        $got = $obj->numberOfSpecialChars( "aaAbcBC" );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 0;
        $got = $obj->numberOfSpecialChars( "abc" );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 3
        $want = 0;
        $got = $obj->numberOfSpecialChars( "AbBCab" );
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/CountTheNumberOfSpecialCharactersII_test.go
/**
 * @version GO 1.26.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestCountTheNumberOfSpecialCharactersII
 */
package GO

import (
	"reflect"
	"testing"
)

func TestCountTheNumberOfSpecialCharactersII(t *testing.T) {
	tables := []struct {
		want int
		low  string
	}{
		{3, "aaAbcBC"},
		{0, "abc"},
		{0, "AbBCab"},
	}

	for _, table := range tables {
		got := numberOfSpecialChars(table.low)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
